### DISCUSSION TITLE
For Direct Forward Pattern, is there a way to return forwarded request while still executing underlying code?

### DISCUSSION THREAD

**Discussion by pkycho92** [dotnet/yarp#1894] [Q&A] [answered]
Hi, i am using Direct Forward Pattern. I need the underlying code to be executed even when the request is forwarded. I tried doing this by calling next(); after the request is forwarded but i am getting a System.InvalidOperationException in System.Private.CoreLib.dll error. Status Code cannot be set because response has already started.
```c#
    app.Use((context, next) =>
    {
            if (conditionIsMet)
            {
                var error = await forwarder.SendAsync(httpContext, endpoint), httpClient, requestOptions, transformer);

                // Check if the proxy operation was successful
                if (error != ForwarderError.None)
                {
                    var errorFeature = httpContext.Features.Get<IForwarderErrorFeature>();
                    var exception = errorFeature.Exception;
                }
            await next(); <-- i am still trying to call the underlying service but return result of forwarded request
            }
            else
            {
                await next();
            }
    });
    app.UseEndpoints(...)
```

I tried instantiating a new thread and calling next() in there but i am getting the same error.

**Comment by Tratcher** [comment, marked as answer]
You shouldn't call next after forwarding, the response has already been sent. What other code are you trying to execute?

**Reply by pkycho92** [reply]
I am trying to execute the code that would normally execute if the request is not forwarded (ex. if next() is called).

**Reply by pkycho92** [reply]
however, i still need the response from the forwarded request to be returned to the client.

**Reply by Tratcher** [reply]
But why? That next code would also produce a response and you can't send two responses back to the client. Is this for some kind of mirroring or testing? See https://github.com/microsoft/reverse-proxy/issues/105.

**Reply by pkycho92** [reply]
Yep looks like it fits the use case for mirroring. I was going towards a similar solution. Clone the request, change the URI and send the request in a fire and forget manner
